### PR TITLE
FIX-#4385: Get rid of `use-deprecated` option in `pip`

### LINKED ISSUE #4385: Get rid of `--use-deprecated=legacy-resolver` option during env installation via `pip`
It's need to get rid of using of deprecated functionality of pip in our scripts since #4365 was merged. The only one place:https://github.com/modin-project/modin/blob/1abb2ea37a21b094688ef666d76ddec0d6fd9eb2/.github/workflows/push-to-master.yml#L118

### PR DESCRIPTION
**Pull request by prutskov**
Signed-off-by: Alexey Prutskov <lehaprutskov@gmail.com>

## What do these changes do?

PR removes using of deprecated functionality of pip during creating of dev env.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4385  
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Comment by prutskov** [comment]
Blocked by #4383

**Comment by prutskov** [comment]
@modin-project/modin-core, PR is ready for review. The changes have been tested in `push-to-master` workflow in this action: https://github.com/modin-project/modin/runs/6019620787?check_suite_focus=true

File: docs/release_notes/release_notes-0.15.0.rst [hunk 1]
```diff
@@ -7,6 +7,7 @@ Key Features and Updates
   * FIX-#4376: Upgrade pandas to 1.4.2 (#4377)
   * FIX-#3615: Relax some deps in development env (#4365)
   * FIX-#4370: Fix broken docstring links (#4375)
+  * FIX-#4385: Get rid from `use-deprecated` option in `pip`` (#4386)
```
Comment (YarShev) [inline docs/release_notes/release_notes-0.15.0.rst:10]:
```suggestion
  * FIX-#4385: Get rid of `use-deprecated` option in `pip`` (#4386)
```

Comment (prutskov) [reply docs/release_notes/release_notes-0.15.0.rst]: fixed

**Review by YarShev — DISMISSED** [review]
LGTM!

**Comment by YarShev** [comment]
@prutskov, please rebase on current master to trigger CI and see proper testing.

**Review by YarShev — APPROVED** [review]

**Pull request merged**